File: first/helloWorld.go
package main

import (
    "fmt"
    "sort"
)

func main() {
    fmt.Println("Hello, World!")

    // Additional lines
    numbers := []int{5, 3, 4, 1, 2}
    fmt.Println("Unsorted numbers:", numbers)

    // Sorting using sort package
    sort.Ints(numbers)
    fmt.Println("Sorted numbers:", numbers)

    // Bubble sort
    bubbleSorted := bubbleSort([]int{5, 3, 4, 1, 2})
    fmt.Println("Bubble sorted numbers:", bubbleSorted)

    // Insertion sort
    insertionSorted := insertionSort([]int{5, 3, 4, 1, 2})
    fmt.Println("Insertion sorted numbers:", insertionSorted)
}

func bubbleSort(arr []int) []int {
    n := len(arr)
    for i := 0; i < n; i++ {
        for j := 0; j < n-i-1; j++ {
            if arr[j] > arr[j+1] {
                arr[j], arr[j+1] = arr[j+1], arr[j]
            }
        }
    }
    return arr
}

func insertionSort(arr []int) []int {
    for i := 1; i < len(arr); i++ {
        key := arr[i]
        j := i - 1
        for j >= 0 && arr[j] > key {
            arr[j+1] = arr[j]
            j = j - 1
        }
        arr[j+1] = key
    }
    return arr
}

File: first/helloWorld.cs
using System;
using System.Linq;

class Program {
    static void Main() {
        Console.WriteLine("Hello, World!");

        int[] numbers = { 5, 3, 8, 4, 2 };
        Console.WriteLine("Original array: " + string.Join(", ", numbers));

        // Bubble Sort
        BubbleSort(numbers);
        Console.WriteLine("Bubble Sorted array: " + string.Join(", ", numbers));

        // Reset array
        numbers = new int[] { 5, 3, 8, 4, 2 };

        // Selection Sort
        SelectionSort(numbers);
        Console.WriteLine("Selection Sorted array: " + string.Join(", ", numbers));

        // Reset array
        numbers = new int[] { 5, 3, 8, 4, 2 };

        // Insertion Sort
        InsertionSort(numbers);
        Console.WriteLine("Insertion Sorted array: " + string.Join(", ", numbers));
    }

    static void BubbleSort(int[] array) {
        int n = array.Length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    // Swap temp and array[i]
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    static void SelectionSort(int[] array) {
        int n = array.Length;
        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (array[j] < array[minIdx]) {
                    minIdx = j;
                }
            }
            // Swap the found minimum element with the first element
            int temp = array[minIdx];
            array[minIdx] = array[i];
            array[i] = temp;
        }
    }

    static void InsertionSort(int[] array) {
        int n = array.Length;
        for (int i = 1; i < n; ++i) {
            int key = array[i];
            int j = i - 1;

            // Move elements of array[0..i-1], that are greater than key, to one position ahead of their current position
            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
    }
}
